packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue

```diff
@@ -35,16 +35,25 @@
     },
     setup(props) {
       const snippetConfig = inject('snippetConfig') as SnippetConfig;
-      const extraParams = ref({});
+      const extraParams = ref<Record<string, unknown>>({});
+      const initialExtraParams = ref<Record<string, unknown>>({});
 
       watch(
         [() => snippetConfig, () => props.values],
         () => {
+          const newExtraParams: Record<string, unknown> = {};
+
           forEach({ ...props.values, ...snippetConfig }, (name, value) => {
             if (!props.excludedExtraParams.includes(name)) {
-              extraParams.value = { ...extraParams.value, [name]: value };
+              newExtraParams[name] = value;
```

Comment:
I think we can do this directly with extraParams ref.

`extraParams.value[name] = value;`

This way we could skip the comparison that follows to check if something have changed. Assigning the same value should not trigger the reactivity in Vue.

Reply:
In that case, `ExtraParams` component won't trigger the reactivity of the prop because it requires an update of the whole object. For example, triggering `InterfaceX.setSnippetConfig({ lang: 'es' })`, changes the `extraParams` value `extraParams.value[name] = value;` but the watcher inside `ExtraParams` component is not triggered. 

Trying to look for a clear solution instead of saving a copy and comparing with, I did that and tested in the X demo:
```TS
const extraParams = ref<Dictionary<unknown>>({});

watch(
  [() => snippetConfig, () => props.values],
  () => {
    const newExtraParams = { ...snippetConfig, ...props.values };
    const shouldChange = Object.entries(newExtraParams).some(
      ([name, value]) =>
        !props.excludedExtraParams.includes(name) && extraParams.value?.[name] !== value
    );
    if (shouldChange) {
      extraParams.value = newExtraParams;
    }
  },
  {
    deep: true,
    immediate: true
  }
);
```

Perhaps I missed something. Tell me WDYT